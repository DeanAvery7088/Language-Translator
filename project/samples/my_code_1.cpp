#include <iostream>
#include "Matrix.h"
#include <math.h>
using namespace std;
int main ( ) {
matrix matrix1 ( matrix::matrixRead("../samples/my_code_1_matrix1.data") ) ; 

cout << matrix1;
cout << "\n";
matrix matrix2 ( matrix::matrixRead("../samples/my_code_1_matrix2.data") ) ; 

cout << matrix2;
cout << "\n";
int numRows;
numRows = matrix1.numRows();

int numCols;
numCols = matrix2.numCols();

matrix newMatrix1 (numRows,numCols);
for (int a = 0; a < numRows; a++) {
for (int b = 0; b < numCols; b++) {
*(newMatrix1.access(a, b)) = (a * b); 
} }

newMatrix1 = (matrix1 * matrix2);

cout << newMatrix1;
cout << "\n";
matrix matrix3 ( matrix::matrixRead("../samples/my_code_1_matrix3.data") ) ; 

cout << matrix3;
cout << "\n";
matrix matrix4 ( matrix::matrixRead("../samples/my_code_1_matrix4.data") ) ; 

cout << matrix4;
cout << "\n";
numRows = matrix3.numRows();

numCols = matrix4.numCols();

matrix newMatrix2 (numRows,numCols);
for (int a = 0; a < numRows; a++) {
for (int b = 0; b < numCols; b++) {
*(newMatrix2.access(a, b)) = (a * b); 
} }

newMatrix2 = (matrix3 * matrix4);

}
