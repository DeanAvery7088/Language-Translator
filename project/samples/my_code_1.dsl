/* Reads two matrices as inputs and prints them to the screen.  Then 
   multiplies the two and prints the resulting matrix.

   This program uses the EXTRA CREDIT matrix multiplication overload.
 */

main ( ) {

// First trying matrix multiplication on compatible matrices
matrix matrix1 = matrixRead( "../samples/my_code_1_matrix1.data" );
print(matrix1);
print ("\n");
matrix matrix2 = matrixRead( "../samples/my_code_1_matrix2.data" );
print(matrix2);
print ("\n");

int numRows;
numRows = numRows(matrix1);
int numCols;
numCols = numCols(matrix2);

matrix newMatrix1 [numRows : numCols] a : b = a * b;

// EXTRA CREDIT:  simply multiply using overloaded *
newMatrix1 = matrix1 * matrix2;

print(newMatrix1);
print ("\n");

// Try matrix multiplication on incompatible matrices
matrix matrix3 = matrixRead( "../samples/my_code_1_matrix3.data" );
print(matrix3);
print ("\n");
matrix matrix4 = matrixRead( "../samples/my_code_1_matrix4.data" );
print(matrix4);
print ("\n");

numRows = numRows(matrix3);
numCols = numCols(matrix4);

matrix newMatrix2 [numRows : numCols] a : b = a * b;

newMatrix2 = matrix3 * matrix4;


}
