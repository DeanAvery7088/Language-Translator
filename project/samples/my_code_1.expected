2 4
1  3  5  7  
2  4  6  8  

4 3
1  8  9  
2  7  10  
3  6  11  
4  5  12  

2 3
50  94  178  
60  120  220  

4 3
1  8  9  
2  7  10  
3  6  11  
4  5  12  

2 4
1  3  5  7  
2  4  6  8  

ERROR: Trying to perform multiplication on incompatible matrices
